pkg/builder/controller.go

```diff
@@ -228,26 +237,33 @@ func (blder *Builder) loadRestConfig() {
 	}
 }
 
-func (blder *Builder) getControllerName() (string, error) {
+func (blder *Builder) getControllerName(gvk schema.GroupVersionKind) string {
 	if blder.name != "" {
-		return blder.name, nil
+		return blder.name
 	}
-	gvk, err := getGvk(blder.forInput.object, blder.mgr.GetScheme())
-	if err != nil {
-		return "", err
-	}
-	return strings.ToLower(gvk.Kind), nil
+	return strings.ToLower(gvk.Kind)
 }
 
 func (blder *Builder) doController(r reconcile.Reconciler) error {
-	name, err := blder.getControllerName()
-	if err != nil {
-		return err
-	}
 	ctrlOptions := blder.ctrlOptions
 	if ctrlOptions.Reconciler == nil {
 		ctrlOptions.Reconciler = r
 	}
-	blder.ctrl, err = newController(name, blder.mgr, ctrlOptions)
+
+	// Retrieve the GVK from the object we're reconciling
+	// to prepopulate logger information, and to optionally generate a default name.
+	gvk, err := getGvk(blder.forInput.object, blder.mgr.GetScheme())
+	if err != nil {
+		return err
+	}
+
+	// Setup the logger.
+	if ctrlOptions.Log == nil {
+		ctrlOptions.Log = blder.mgr.GetLogger()
+	}
+	ctrlOptions.Log = ctrlOptions.Log.WithValues("groupVersionKind", gvk.String())
```

Comment:
How do we properly distinguish this from what we do in the encoder https://github.com/kubernetes-sigs/controller-runtime/blob/6633846423ba4853b1c4466f28aa6b66d6804930/pkg/log/zap/kube_helpers.go#L69 ?

I think the log key should maybe be `reoncilerGroupVersionKind` or something like that?

Reply:
It seems like the logging here should remove the need to enrich the log entries in the zap helper. Would you expect any delta between the gvk defined here and the resource passed to reconcile?

Reply:
Its totally possible that during the reconciliation of an object type a an object type b has to be updated and that fails so I think the two are complementary.

Reply:
Fixed, used `reconcilerGroup` and `reconcilerKind`